evalml/pipelines/time_series_regression_pipeline.py

```diff
@@ -75,7 +75,10 @@ def fit(self, X, y):
 
         y_shifted = y.shift(-self.gap)
         X_t, y_shifted = drop_rows_with_nans(X_t, y_shifted)
-        self.estimator.fit(X_t, y_shifted)
+        if self.estimator is not None:
```

Comment:
The logic here and in the other time series classification pipelines makes a greater case for https://github.com/alteryx/evalml/issues/2614 :)